🚀 1단계 - 용어 사전 만들기

writer0713 opened the pull request:
안녕하세요, 멘토님. 
페어와 `🚀 1단계 - 용어 사전 만들기` 토론하여 용어 사전 만들었습니다.
이번 pr 도 잘 부탁드립니다.

아래는 페어와 진행하다가 공통적으로 궁금한점이 생겨 질문 드립니다.

- '상품 수량' 이나 '주문 항목 수량' 과 같이 중복되는 용어의 경우, 전략적 설계에서도 bounded context 를 나눠서 각각 작성해야 하나요?
- '메뉴 이름' 이나 '메뉴 가격' 처럼 너무 당연해서 설명란에 쓸게 없는 것들도 용어집에 작성해야 하나요?

writer0713 committed:
docs: 용어사전 정리

writer0713 committed:
docs: README.md 에 용어집 수정

- 설명 가다듬기

writer0713 committed:
docs: README.md 용어집 수정

- 설명이 어색한 부분 수정

writer0713 committed:
docs: 용어 사전 수정
- 페어와 토론하여 용어 사전 수정

writer0713 committed:
docs: README.md 오타 제거

HackSung commented on README.md:
```diff
@@ -96,8 +96,81 @@ docker compose -p kitchenpos up -d
+| 한글명   | 영문명              | 설명                                       |
+|-------|------------------|------------------------------------------|
+| 상품    | Product          | 키친포스에 판매되고 있거나 판매될 음식과 관련된 여러가지 정보를 말한다. |
+| 상품 등록 | Register Product | 상품을 키친포스에 등록하여 해당 상품을 메뉴에 추가할 수 있습니다.    |
```
다른 용어의 설명은 평서문체인데 여기는 존댓말 설명이네요?
일관성 있게 맞춰보는건 어떨까요?

writer0713 replied:
일관성 있게 평서문체로 맞추겠습니다 :)

HackSung commented on README.md:
```diff
@@ -96,8 +96,81 @@ docker compose -p kitchenpos up -d
+| 배달 대행사    | DeliveryAgency    | 배달을 전문으로 하는 외부 대행사. 호출을 받으면 배달 대행사에서 배달을 대행해준다.                                               |
+| 배달 대행사 호출 | Call Delivery     | 손님에게 배달을 하기 외부 배달 대행사에게 요청하는 행위                                                               |
+
+### 기타
```
여러 도메인이 공통으로 사용하는 용어를 `기타`로 분류하신 것 같은데 `기타`보다는 `공통`이 어떠세요?
저의 경우는 내용이 동일해 보여도 해당 용어를 사용하고 있는 도메인쪽에 각각 기술해주는걸 선호합니다. 😄 
정환님 생각은 어떠세요? 🤔

writer0713 replied:
`기타` 와 `공통` 을 고민하다 `기타` 로 했었는데,
그보다는 말씀하신것처럼 각 도메인 영역에 각각 추가해주는 방식이 나을것 같습니다 :)

HackSung commented on README.md:
```diff
@@ -96,8 +96,81 @@ docker compose -p kitchenpos up -d
+| 주문한 메뉴   | Ordered Menu      | 손님이 주문한 메뉴를 뜻한다.                             |
+| 등록된 메뉴   | Registered Menu   | 점장이 키친포스에 등록한 메뉴를 뜻한다.                       |
+| 숨겨진 메뉴   | Hidden Menu       | 키친포스에서 숨겨진 메뉴를 뜻한다.                          |
+| 메뉴 등록    | Register Menu     | 한 개 이상의 등록된 상품으로 메뉴를 등록할 수 있습니다.             |
```
의미를 좀 더 명확하게 하기 위해 아래 표현으로 약간 수정해보았는데 어떠세요?
```suggestion
| 메뉴 등록    | Register Menu     | 한 개 이상의 등록된 상품으로 구성된 메뉴를 등록할 수 있다.             |
```

writer0713 replied:
오, 훨씬 매끄러운것 같습니다 :)

HackSung commented on README.md:
```diff
@@ -96,8 +96,81 @@ docker compose -p kitchenpos up -d
+| 등록된 메뉴   | Registered Menu   | 점장이 키친포스에 등록한 메뉴를 뜻한다.                       |
+| 숨겨진 메뉴   | Hidden Menu       | 키친포스에서 숨겨진 메뉴를 뜻한다.                          |
+| 메뉴 등록    | Register Menu     | 한 개 이상의 등록된 상품으로 메뉴를 등록할 수 있습니다.             |
+| 메뉴 노출    | Display Menu      | 메뉴를 노출할 수 있다. 손님은 노출된 메뉴를 주문할 수 있다.          |
```
좀 더 이해하기 쉽게 아래 표현으로 약간 다듬어 보았는데 어떠세요?
```suggestion
| 메뉴 노출    | Display Menu      | 메뉴를 손님에게 노출할 수 있으며, 손님은 노출된 메뉴만 주문할 수 있다.          |
```

writer0713 replied:
훨씬 매끄러운것 같습니다..!

HackSung commented on README.md:
```diff
@@ -96,8 +96,81 @@ docker compose -p kitchenpos up -d
+| 숨겨진 메뉴   | Hidden Menu       | 키친포스에서 숨겨진 메뉴를 뜻한다.                          |
+| 메뉴 등록    | Register Menu     | 한 개 이상의 등록된 상품으로 메뉴를 등록할 수 있습니다.             |
+| 메뉴 노출    | Display Menu      | 메뉴를 노출할 수 있다. 손님은 노출된 메뉴를 주문할 수 있다.          |
+| 메뉴 숨김    | Hide Menu         | 점장은 메뉴를 숨길수 있다. 메뉴를 숨기면 손님은 해당 메뉴를 주문할 수 없다. |
```
의미를 좀 더 명확하게 하기 위해 아래 표현으로 약간 다듬어 보았는데 어떠세요?
```suggestion
| 메뉴 숨김    | Hide Menu         | 점장은 메뉴를 숨길 수 있으며, 숨겨진 메뉴는 고객이 주문할 수 없다. |
```

writer0713 replied:
훨씬 나은거 같습니다 :)

HackSung commented on README.md:
```diff
@@ -96,8 +96,81 @@ docker compose -p kitchenpos up -d
+| 주문 테이블 이름   | Table Name              | 테이블의 이름                   |
+| 빈테이블        | Empty Table             | 손님이 앉지 않은 빈 테이블           |
+| 빈테이블 해지     | Sit Table               | 손님이 주문 테이블에 앉은 상태를 표현한다.  |
+| 빈테이블 설정     | Clear Table             | 주문이 완료된 손님의 테이블을 정리한다.    |
```
`빈테이블 해지`나 `빈테이블 설정`이라는 용어가 뭔가 부자연스러워 보이고 영문명과 맞지 않아 보이는데 `테이블 착석`, `테이블 이용`, `테이블 점유`, `테이블 비우기` 같은 표현을 써보는 건 어떨까요?

HackSung replied:
위 제안은 확인해 보셨을까요? 🤔

HackSung replied:
https://github.com/next-step/ddd-strategic-design/pull/350#discussion_r1604220638
> 빈테이블 해지나 빈테이블 설정이라는 용어가 뭔가 부자연스러워 보이고 영문명과 맞지 않아 보이는데 테이블 착석, 테이블 이용, 테이블 점유, 테이블 비우기 같은 표현을 써보는 건 어떨까요?

위 제안을 확인하셨다고 이모지를 다셨지만 반영하지 않으신 이유를 혹시 나눠 주실 수 있을까요? 😄

writer0713 replied:
@writer0713 
> 빈테이블 해지나 빈테이블 설정이라는 용어가 뭔가 부자연스러워 보이고 영문명과 맞지 않아 보이는데 테이블 착석, 테이블 이용, 테이블 점유, 테이블 비우기 같은 표현을 써보는 건 어떨까요?
에고.. 제가 다른 부분을 수정하고 push 하면서 `Outdated` 가 되어 첫 댓글이 안보였었네요.

writer0713 replied:
@writer0713 
> 빈테이블 해지나 빈테이블 설정이라는 용어가 뭔가 부자연스러워 보이고 영문명과 맞지 않아 보이는데 테이블 착석, 테이블 이용, 테이블 점유, 테이블 비우기 같은 표현을 써보는 건 어떨까요?

에고.. 제가 다른 부분을 수정하고 push 하면서 `Outdated` 가 되어 첫 댓글이 안보였었네요.
아래와 같이 수정하겠습니다 :) 
- 빈테이블 해지 =>  테이블 착석
- 빈테이블 설정 =>  테이블 정리

![스크린샷 2024-05-18 오후 6 02 49](https://github.com/next-step/ddd-strategic-design/assets/5811400/9cbef829-eb8f-4d03-8540-f805329d7f06)

HackSung commented on README.md:
```diff
@@ -96,8 +96,81 @@ docker compose -p kitchenpos up -d
+| 빈테이블        | Empty Table             | 손님이 앉지 않은 빈 테이블           |
+| 빈테이블 해지     | Sit Table               | 손님이 주문 테이블에 앉은 상태를 표현한다.  |
+| 빈테이블 설정     | Clear Table             | 주문이 완료된 손님의 테이블을 정리한다.    |
+| 방문한 손님 수    | Number Of Guests        | 주문 테이블을 사용하는 손님 수         |
```
방문은 했지만 테이블을 이용하지 않을 수도 있으니 `착석한 손님 수` 같이 명확하게 표현해 보는 건 어떠세요?

HackSung replied:
위 제안은 확인해 보셨을까요? 🤔

HackSung replied:
https://github.com/next-step/ddd-strategic-design/pull/350#discussion_r1604221206
> 방문은 했지만 테이블을 이용하지 않을 수도 있으니 착석한 손님 수 같이 명확하게 표현해 보는 건 어떠세요?

위 제안을 확인하셨다고 이모지를 다셨지만 반영하지 않으신 이유를 혹시 나눠 주실 수 있을까요? 😄

writer0713 replied:
@HackSung 
제가 다른 부분을 수정하고 push 하면서 Outdated 가 되어 첫 댓글이 안보였었네요.
`착석한 손님 수` 가 훨씬 나은 표현 같습니다 :)

HackSung commented on README.md:
```diff
@@ -96,8 +96,81 @@ docker compose -p kitchenpos up -d
+| 한글명      | 영문명               | 설명                                           |
+|----------|-------------------|----------------------------------------------|
+| 메뉴       | Menu              | 한 개 이상의 상품으로 이루어진 상품의 집합을 의미한다.              |
+| 주문한 메뉴   | Ordered Menu      | 손님이 주문한 메뉴를 뜻한다.                             |
```
사소하지만 설명에서 `의미한다`와 `뜻한다`라는 동사가 혼용되고 있는데 통일시켜 일관성 있게 만드는건 어떠세요?

writer0713 replied:
그러네요. `의미한다` 로 통일시키겠습니다 :)

HackSung commented on README.md:
```diff
@@ -96,8 +96,81 @@ docker compose -p kitchenpos up -d
+| 배달 완료     | Complete Delivery | 손님에게 정상적으로 음식을 배달 완료한 상태                                                                      |
+| 완료        | Complete          | 손님에게 음식 제공이 끝나서 주문이 완료된 상태                                                                    |
+| 배달 대행사    | DeliveryAgency    | 배달을 전문으로 하는 외부 대행사. 호출을 받으면 배달 대행사에서 배달을 대행해준다.                                               |
+| 배달 대행사 호출 | Call Delivery     | 손님에게 배달을 하기 외부 배달 대행사에게 요청하는 행위                                                               |
```
```suggestion
| 배달 대행사 호출 | Call Delivery     | 손님에게 배달을 하기 위해 외부 배달 대행사에게 요청하는 행위                                                               |
```

HackSung commented on README.md:
```diff
@@ -96,8 +96,81 @@ docker compose -p kitchenpos up -d
+| 주문 유형     | OrderType         | 손님이 주문할 수 있는 주문 방식을 의미한다. 주문 유형에는 배달, 포장, 매장이 있다.                                             |         
+| 배달 주문     | Delivery Order    | 손님이 주문을 하면 배달 대행업체를 통해 배달되는 주문                                                                |
+| 포장 주문     | TakeOut Order     | 손님이 주문을 하고 음식이 준비되면 직접 찾으러 가는 주문                                                              |
+| 매장 주문     | EatIn Order       | 손님이 매장에서 직접하는 주문                                                                              |
```
포장 주문도 손님이 매장에서 직접 주문할 수 있으니 좀 더 조건을 넣어 명확하게 설명 해보는 건 어떨까요?
```suggestion
| 매장 주문     | EatIn Order       | 손님이 매장내에서 식사하기 위에 직접하는 주문                                                                              |
```

writer0713 replied:
@HackSung 
아, 요 부분 안그래도 페어분이랑 얘기가 나왔었는데요.
요구사항이랑 로직상에서는 `매장내에서 식사 하는 행위` 에 대해서는 설명이나 구현이 없더라구요.

테이블의 경우도, 손님이 식사를 마치면 치우는게 아니라 주문이 완료상태가 되면 바로 치울수 있도록 구현이 되어있어서 저희가 알고 있는 일반적인 요구사항이 아닌가 라는 생각을 했었습니다.
제가 잘못 이해한걸까요?

제가 잘못 이해한거라면 제안 주신것처럼 `매장내에서 식사하기 위해` 라는 설명이 들어는게 더 나은것 같긴합니다 :)

HackSung replied:
> 한가지, 매장 주문 에 대해 피드백 주신 부분은 제가 이해를 잘못한건지 궁금하여 위에 질문 남겼습니다 :)

주문 유형은 주문 생성 시 필수 입력 항목이라 
점장이 손님에게 포장주문인지 매장주문인지 먼저 물어보고 주문을 생성하게 될 것 같아요.
주문 유형이 결정되고 나면 주문 유형에 따른 프로세스가 진행되겠죠?
결정된 주문 유형에 대해 명확한 상황설명을 넣어 보면 좋을 것 같아 제안 드린 거 였어요. 😄 
그런데 주문완료처리를 현실세계에 대입해 본 적은 없었는데
생각해보니 로직대로라면 현실세계에서는 테이블을 완전히 치웠는지 점장이 확인하고 주문 완료 처리를 해야 겠네요? ㅎㅎ

writer0713 replied:
@HackSung 
일반적으로 `로직` 보다는 `기획 (혹은 이벤트 스토밍)` 을 먼저 진행하니 말씀하신대로 `매장내에서 식사하기 위해` 라는 설명을 추가하겠습니다 :)

HackSung commented:
> '상품 수량' 이나 '주문 항목 수량' 과 같이 중복되는 용어의 경우, 전략적 설계에서도 bounded context 를 나눠서 각각 작성해야 하나요?

` 상품 수량`과 `주문 항목 수량`이라는 용어가 각각 다른 context에서 다른 의미를 가진다면 이를 분리하여 정의하는 것이 좋습니다.
그런데 이미 도메인을 상품과 주문으로 나누신 것 같은데 각각의 도메인에 나눠서 작성 가능하신 상황 아닌가요? 🤔 

> '메뉴 이름' 이나 '메뉴 가격' 처럼 너무 당연해서 설명란에 쓸게 없는 것들도 용어집에 작성해야 하나요?

용어 사전의 목적은 도메인의 주요 개념을 명확하게 정의하고, 팀원들이 동일한 언어를 사용하여 의사소통하는 것입니다. 
따라서 `메뉴 이름`이나 `메뉴 가격`과 같이 당연하다고 생각될 수 있는 용어도
팀 내에서 공통된 이해를 가지고 있을 수 있도록 용어 사전에 포함시키는 것이 좋습니다. 
설명란이 비어있어도 해당 용어가 도메인에서 어떤 역할을 하는지를 나타내는 것 자체가 중요할 것 같습니다. 😄

HackSung commented on README.md:
```diff
@@ -96,8 +96,81 @@ docker compose -p kitchenpos up -d
+| 한글명       | 영문명               | 설명                                                                                            |
+|-----------|-------------------|-----------------------------------------------------------------------------------------------|
+| 주문        | Order             | 키친포스에 있는 메뉴를 손님이 구매한 것을 의미한다.                                                                 |
+| 주문 유형     | OrderType         | 손님이 주문할 수 있는 주문 방식을 의미한다. 주문 유형에는 배달, 포장, 매장이 있다.                                             |         
```
주문 유형의 나열과 각각의 주문 유형에 대한 정의 훌륭합니다. 👍 👍

HackSung commented on README.md:
```diff
@@ -96,8 +96,81 @@ docker compose -p kitchenpos up -d
+| 매장 주문     | EatIn Order       | 손님이 매장에서 직접하는 주문                                                                              |
+| 배달 주소     | DeliveryAddress   | 손님이 음식을 받을 주소를 뜻한다. `배달 주문` 의 경우 필요하다.                                                        |
+| 주문 등록     | Create Order      | 손님이 키친포스에 있는 메뉴를 구매하는 행위를 의미한다.                                                               |
+| 주문 상태     | OrderStatus       | 주문이 받아들여지고 나서부터 완료될 때까지의 상태를 나타낸다. 주문 상태는 `접수 대기`, `접수`, `서빙`, `배달 시작`, `배달 완료`, `완료` 로 구성된다. |
```
주문 상태의 나열과 각각의 주문 상태에 대한 설명 너무 좋습니다. 💯

HackSung reviewed (changes requested):
안녕하세요~ 정환님!
이번 미션을 함께 하게 되어 반갑습니다. 😄 
도메인별로 용어 사전을 잘 정리해 주신 것 같은데 피드백을 달았으니 확인 부탁드려요~ 🙏

writer0713 committed:
docs: 평서문으로 수정

writer0713 committed:
docs: `기타` 에 기술한 용어들을 각 도메인에 추가

writer0713 committed:
docs: '메뉴 등록' 설명 수정

- 좀 더 명확한 의미를 가지도록 수정

Co-authored-by: HackSung <spfedit@gmail.com>

writer0713 committed:
docs: `메뉴 노출` 에 대한 설명 수정

- 좀 더 매끄럽게 수정

Co-authored-by: HackSung <spfedit@gmail.com>

writer0713 committed:
docs: '메뉴 숨김' 에 대한 설명 수정

- 더 명확한 표현으로 수정

Co-authored-by: HackSung <spfedit@gmail.com>

writer0713 committed:
docs: '의미한다' 라는 표현으로 통일
- '의미한다' 와 '뜻한다' 라는 표현이 혼용되고 있음
- '의미한다' 로 통일

writer0713 committed:
docs: '배달 대행사 호출' 설명 수정
- 좀 더 매끄럽게 수정

writer0713 committed:
docs: '의미한다' 로 통일
- '뜻한다', '나타낸다' 등을 '의미한다' 로 통일

writer0713 committed:
docs: '주문' 도메인에 '주문 항목 수량' 추가

writer0713 commented:
@HackSung 
꼼꼼한 피드백 감사합니다 :) 
피드백 주신 부분들 대부분 동의하여 수정했습니다.

한가지, `매장 주문` 에 대해 피드백 주신 부분은 제가 이해를 잘못한건지 궁금하여 위에 질문 남겼습니다 :) 
> | 매장 주문     | EatIn Order       | 손님이 `매장내에서 식사하기 위해` 직접하는 주문

HackSung commented on README.md:
```diff
@@ -96,8 +96,80 @@ docker compose -p kitchenpos up -d
+| 주문 테이블 목록   | Tables                  | 복수개의 주문 테이블               |
+| 손님          | Customer                | 음식점에 음식을 주문하는 고객          |
+
+### 주문
```
이대로도 훌륭해 보이지만,
주문 도메인에 특정 주문유형에서만 사용하는 용어들이 섞여 있는 상황인데
수업시간에 배우신 Bounded Context개념을 적용하여
주문 공통과 주문 유형별로 나눠진 용어사전 작성을 시도해 보는 건 어떠세요? 🤔
정환님의 의견은 어떠한지 굼긍합니다.
굳이 나눌 필요 없다고 판단되시면 이대로 진행하셔도 무방합니다. 😄

writer0713 replied:
특정 주문유형에 대한 용어사전을 만들면 그저 용어들이 중복 생성되는게 아닌가 생각이 들었었는데요.
각 주문유형에 대한 용어사전을 만들다보니 아래와 같은 장점이 있는것 같다는 생각이 들었습니다.
- `한글명/영문명` 은 동일해도 `설명` 은 해당 주문유형과 관련하여 좀 더 세부적으로 나타낼수 있다.
- 해당 주문유형에서만 사용되는 용어들 (ex. 테이블 관련) 이 추가되기 때문에 해당 주문유형에 대해 파악하기가 좀 더 수월하다.

HackSung replied:
정환님의 소중한 의견을 나눠주셔서 감사합니다. 🙏 

주문 유형별로 단어와 의미가 동일한 용어들이 많다면 처음에 정환님이 하셨던 것 처럼 
따로 뽑아 공통으로 분류해도 좋을 것 같다는 생각이 저도 들었습니다. 😄 

주문 유형별로 비즈니스 로직이 크게 다르다면, 
각 주문 유형을 별도의 Bounded Context로 분리하는 것이 좋을 수 있고
이렇게 하면 각 주문 유형에 대한 복잡성을 독립적으로 관리할 수 있는 장점이 있을 것 같습니다.
반면에 주문 유형별로 비즈니스 로직이 크게 다르지 않고 주문 처리 과정의 대부분이 공통적이라면
모든 주문 유형을 하나의 Bounded Context 내에서 관리하는 것이 더 효율적일 수 있을 것 같습니다.

결국 정답이 정해져 있는 건 아니니 
주문 유형별로 Bounded Context를 나누는 것이 바람직한지는 주문 처리 과정의 비즈니스 로직, 복잡성, 공통성 등을 고려하여 
결정해야 할 것 같고 이는 프로젝트의 특성과 요구사항에 따라 달라질 수 있으므로
팀과 함께 충분히 논의하고 결정하는 것이 중요하다고 생각되네요. 😄

HackSung replied:
정환님의 소중한 의견을 나눠주셔서 감사합니다. 🙏 

작성해 주신 주문 유형별 용어사전에서
주문 유형별로 단어와 의미가 중복되는 용어들이 많은 걸 보니 
처음에 정환님이 하셨던 것 처럼 따로 뽑아 주문 도메인 공통으로 분류해도 좋을 것 같다는 생각이 드네요 😅  

Bounded Context에 대한 저의 생각도 나눠 보고 싶은데요,
주문 유형별로 비즈니스 로직이 크게 다르다면, 
각 주문 유형을 별도의 Bounded Context로 분리하는 것이 좋을 수 있고
이렇게 하면 각 주문 유형에 대한 복잡성을 독립적으로 관리할 수 있는 장점이 있을 것 같습니다.
반면에 주문 유형별로 비즈니스 로직이 크게 다르지 않고 주문 처리 과정의 대부분이 공통적이라면
모든 주문 유형을 하나의 Bounded Context 내에서 관리하는 것이 더 효율적일 수 있을 것 같습니다.

결국 정답이 정해져 있는 건 아니니 
주문 유형별로 Bounded Context를 나누는 것이 바람직한지는 주문 처리 과정의 비즈니스 로직, 복잡성, 공통성 등을 고려하여 
결정해야 할 것 같고 이는 프로젝트의 특성과 요구사항에 따라 달라질 수 있으므로
팀과 함께 충분히 논의하고 결정하는 것이 중요하다고 생각되네요. 😄

HackSung reviewed (changes requested):
피드백 잘 반영해 주셨습니다. 👍 
몇가지 추가 질문드린 게 있는데 확인 부탁드릴께요 🙏

writer0713 committed:
docs: '매장 주문' 에 대한 설명 수정
- 좀 더 명확한 상황설명 추가

writer0713 committed:
docs: 주문 유형별 용어사전 추가
- 배달 주문 용어사전 추가
- 포장 주문 용어사전 추가
- 매장 주문 용어사전 추가

HackSung reviewed (changes requested):
용어 사전 만들기 미션의 제 마지막 생각과 질문을 몇개 남겨 드렸는데 
확인하시고 PR주시면 바로 머지하겠습니다. 😄

writer0713 commented:
> 위 제안을 확인하셨다고 이모지를 다셨지만 반영하지 않으신 이유를 혹시 나눠 주실 수 있을까요? 😄

제가 README.md 파일을 수정하고 push 하면서 남겨주신 첫번째 댓글이 안보였었네요 ^^;
다시 확인하여 수정했습니다!

> 정환님의 소중한 의견을 나눠주셔서 감사합니다. 🙏
작성해 주신 주문 유형별 용어사전에서
주문 유형별로 단어와 의미가 중복되는 용어들이 많은 걸 보니
처음에 정환님이 하셨던 것 처럼 따로 뽑아 주문 도메인 공통으로 분류해도 좋을 것 같다는 생각이 드네요 😅
Bounded Context에 대한 저의 생각도 나눠 보고 싶은데요,
주문 유형별로 비즈니스 로직이 크게 다르다면,
각 주문 유형을 별도의 Bounded Context로 분리하는 것이 좋을 수 있고
이렇게 하면 각 주문 유형에 대한 복잡성을 독립적으로 관리할 수 있는 장점이 있을 것 같습니다.
반면에 주문 유형별로 비즈니스 로직이 크게 다르지 않고 주문 처리 과정의 대부분이 공통적이라면
모든 주문 유형을 하나의 Bounded Context 내에서 관리하는 것이 더 효율적일 수 있을 것 같습니다.
결국 정답이 정해져 있는 건 아니니
주문 유형별로 Bounded Context를 나누는 것이 바람직한지는 주문 처리 과정의 비즈니스 로직, 복잡성, 공통성 등을 고려하여
결정해야 할 것 같고 이는 프로젝트의 특성과 요구사항에 따라 달라질 수 있으므로
팀과 함께 충분히 논의하고 결정하는 것이 중요하다고 생각되네요. 😄

말씀하신것처럼 현재 `주문` 도메인의 경우는 유형별로 프로세스가 큰 차이가 있지 않아서 굳이 나누지 않아도 될것 같다는 생각이 듭니다.

예를 들어, `매장 주문` <=> `주문 테이블` 과 같이 유형별 주문이 다른 도메인과 엮이는 케이스가 많아진다면 그때는 분리를 하는것도 괜찮을것 같습니다.

아직 수업을 다 듣지 않아서 `aggregate` 나 `domain service` 같은 개념을 대입시켜보지 못했는데, 유형별 주문이 다른 도메인과 엮이고 `상태변화` 가 한 트랜잭션 안에서 이루어져야 한다면 하나의 `aggregate` 로 볼수있을듯 합니다.

꼼꼼하게 리뷰 해주셔서 감사합니다!!

writer0713 committed:
docs: 주문 및 주문테이블 관련 용어사전 수정
- 빈테이블 해지 => 테이블 착서
- 빈테이블 정리 => 테이블 정리
- 방문한 손님 수 => 착석한 손님 수
- 유형별 주문 (배달주문 / 매장주문 / 포장주문) 용어사전 삭제하고 (공통) 주문 용어사전만 남김

writer0713 committed:
docs: 주문 용어사전 수정
- '배달 시작', '배달 완료' 를 상단으로 이동 (주문 상태끼리 모아놓기)

HackSung reviewed (approved):
용어 사전 만들기 미션 잘 진행해 주셨습니다. 👍 
새로 만든 용어 사전을 기반으로 다음 단계 미션을 시작하시면 될 것 같아요! 🚀

writer0713 commented:
@HackSung 
안녕하세요, 리뷰어님!
리뷰해주신것 토대로 페어분과 다시 한번 논의를 하면서 궁금한점이 생겨 질문 남깁니다.

페어의 경우, 설명란에 적을게 없는 경우에는 `정책` 을 적으라는 리뷰를 받았다고 하는데요.
ex) 상품 가격: “상품의 가격은 0보다 작으면 안된다”라는 정책 추가

이 부분에 대해서 개인적인 생각은 아래와 같습니다.
- `요구사항` 에 정책들이 이미 정리가 되어 있는데 `설명` 에 또 적을 필요가 있을까?
- 만약 적는게 낫다고 한다면, `정책` 이 많은 (복수개) 경우 에는 `설명` 란에 이 `정책` 들을 다 적어야 하나?
  - 다 적으면 `요구사항` 과 중복되지 않나?
- 적을 설명이 없을만큼 간단한 용어의 경우, 정책을 적기보다는 그냥 빈칸으로 냅두는게 나을것 같다.

혹시 리뷰어님의 생각은 어떠신지 궁금합니다 :)
감사합니다.

HackSung commented:
안녕하세요 정환님!
수동적인 교육 자세를 벗어나 주말에도 미션에 대해 계속 고민하시는 모습에 열의가 느껴집니다. 🔥 

용어사전 설명란에 적을게 없으면 정책을 적어 주는게 좋다는 리뷰어님의 말씀도 맞아보이고
그에 따른 정환님의 생각도 충분히 일리 있어 보입니다.
물론 저의 생각도 정답은 아니나 정환님의 의문에 조금이나마 도움을 드리고자 제 의견을 남겨보겠습니다. 😄

> 요구사항에 정책들이 이미 정리가 되어 있는데 설명 에 또 적을 필요가 있을까?

저는 정책을 요구사항과 용어사전에 모두 적을 수 있다고 생각합니다.
두 곳에 정책을 같이 추가하는 것은 중복된다고 느낄 수 있지만 
요구사항과 용어사전은 서로 다른 목적을 가지고 있어 동일한 정보라도 다른 역할을 할 수 있을 것 같습니다.
요구사항의 정책은 시스템이 어떻게 동작해야 하는지를 명시하여 시스템의 기능적인 측면을 설명하며 
개발자는 해당 정책을 코드로 구현하는 방법을 결정할 수 있습니다.
반면 용어 사전에서는 `상품 가격`이라는 용어의 개념이 어떤 규칙을 가지는지 도메인의 개념적인 측면을 설명합니다.

> 만약 적는게 낫다고 한다면, 정책 이 많은 (복수개) 경우 에는 설명 란에 이 정책 들을 다 적어야 하나?
> 다 적으면 요구사항 과 중복되지 않나?

관련 정책이 많은 경우에는 모든 정책을 용어사전 설명란에 적을 필요는 없을 것 같습니다. 
중요한 정책이나 핵심적인 비즈니스 규칙만 요약하여 적고 상세한 내용은 아래처럼 설명란에 적어
요구사항 문서에서 참조하게 하는 것도 한 가지 방법이 될 수 있을 것 같습니다.
`나머지 상세 정책은 요구사항 문서에 명시됨`

> 적을 설명이 없을만큼 간단한 용어의 경우, 정책을 적기보다는 그냥 빈칸으로 냅두는게 나을것 같다.

정말로 설명할 내용이 거의 없는 간단한 용어를 빈칸으로 두는 것보다는 최소한의 정보라도 제공하는 것이 좋을 것 같습니다.
시간이 지나면서 새로운 개념이나 정책이 추가될 수 도 있고 문서의 일관성 측면에서도 보기가 좋을 것 같습니다

그럼 개발의 행복을 만끽하시는 일주일 되시길 바랄께요~ 😄
